modules/util/lang-sanitizer/src/main/java/com/liferay/lang/sanitizer/LangSanitizer.java

```diff
@@ -0,0 +1,278 @@
+import java.nio.file.attribute.BasicFileAttributes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import org.owasp.validator.html.Policy;
+import org.owasp.validator.html.PolicyException;
+import org.owasp.validator.html.ScanException;
+
+/**
+ * @author Seiphon Wang
+ */
+public class LangSanitizer {
+
+	public static void main(String[] args) throws Exception {
+		Map<String, String> arguments = ArgumentsUtil.parseArguments(args);
+
+		LangSanitizerArges langSanitizerArges = new LangSanitizerArges();
+
+		String baseDir = GetterUtil.getString(
+			arguments.get("source.base.dir"), LangSanitizerArges.BASE_DIR_NAME);
+
+		langSanitizerArges.setBaseDirName(baseDir);
+
+		LangSanitizer langSanitizer = new LangSanitizer(langSanitizerArges);
+
+		long startTime = System.currentTimeMillis();
+
+		langSanitizer.sanitize();
+
+		long endTime = System.currentTimeMillis();
+
+		_printScanResult(startTime, endTime);
+	}
+
+	public LangSanitizer(LangSanitizerArges langSanitizerArges) {
+		ClassLoader classLoader = LangSanitizer.class.getClassLoader();
+
+		URL antiSamyURL = classLoader.getResource("antisamy-liferay.xml");
+
+		String antsamyPath = antiSamyURL.getFile();
+
+		Policy policy = null;
+
+		try {
+			policy = Policy.getInstance(antsamyPath);
+		}
+		catch (PolicyException policyException) {
+			policyException.printStackTrace();
+		}
+
+		_langSanitizerArges = langSanitizerArges;
+
+		_policy = policy;
+	}
+
+	public List<File> getSantizedFiles() {
+		return _sanitizedFiles;
+	}
+
+	public void sanitize() throws Exception {
+		List<File> fileList = _getAllLanguageProperties(
+			_langSanitizerArges.getBaseDirName());
+
+		ExecutorService executorService = Executors.newFixedThreadPool(10);
+
+		List<Future<List<LangSanitizerMessage>>> futures =
+			new CopyOnWriteArrayList<>();
+
+		for (File file : fileList) {
+			Future<List<LangSanitizerMessage>> future = executorService.submit(
+				new Callable<List<LangSanitizerMessage>>() {
+
+					@Override
+					public List<LangSanitizerMessage> call() throws Exception {
+						return _sanitizeProperites(file);
+					}
+
+				});
+
+			futures.add(future);
+		}
+
+		for (Future<List<LangSanitizerMessage>> future : futures) {
+			_langSanitizerMessages.addAll(future.get());
+		}
+
+		executorService.shutdown();
+
+		while (!executorService.isTerminated()) {
+			Thread.sleep(20);
+		}
+	}
+
+	public LangSanitizerMessage sanitizeContent(
+			File file, String key, String originalValue)
+		throws PolicyException {
+
+		if (key.equals("form-navigator-entry-keys-help")) {
+			return null;
+		}
+
+		String sanitizedValue = originalValue;
+		String value = originalValue;
+
+		try {
```

Comment:
I think we should keep this try cache since the ScanException will terminate the process